### PR TITLE
PYTHON-1936 Remove listCollections check from Database.create_collection

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: doc/migrate-to-pymongo4.rst [hunk 1]
```diff
@@ -330,6 +330,15 @@ Can be changed to this::
...
 You must now explicitly compare with None.
 
+Database.create_collection raises OperationFailure
+..................................................
+:meth:`~pymongo.database.Database.create_collection` now raises an
+:class:`~pymongo.errors.OperationFailure` exception instead of a
+:class:`~pymongo.errors.CollectionInvalid` exception when a collection
+already exists. Any code that relies on catching a
+:class:`~pymongo.errors.CollectionInvalid` exception can simply catch
+:class:`~pymongo.errors.OperationFailure` instead.
```
Comment (ShaneHarvey) [inline doc/migrate-to-pymongo4.rst:340]: Given the behavioral differences between different clusters on different server versions, is this statement true?

File: test/test_database.py [hunk 2]
```diff
@@ -132,7 +131,8 @@ def test_create_collection(self):
         db = Database(self.client, "pymongo_test")
 
         db.test.insert_one({"hello": "world"})
-        self.assertRaises(CollectionInvalid, db.create_collection, "test")
+        if not client_context.is_topology_type(["sharded"]):
```
Comment (ShaneHarvey) [inline test/test_database.py:134]: `client_context.is_topology_type(["sharded"])` -> `client_context.is_mongos`

It's the same thing, the second is just less verbose.

File: test/test_database.py [hunk 3]
```diff
@@ -148,7 +148,8 @@ def test_create_collection(self):
         db.drop_collection("test.foo")
         db.create_collection("test.foo")
         self.assertTrue("test.foo" in db.list_collection_names())
-        self.assertRaises(CollectionInvalid, db.create_collection, "test.foo")
+        if not client_context.is_topology_type(["sharded"]):
```
Comment (ShaneHarvey) [inline test/test_database.py:151]: `client_context.is_topology_type(["sharded"])` -> `client_context.is_mongos`

File: test/test_transactions.py [hunk 4]
```diff
@@ -233,9 +232,10 @@ def create_and_insert(session):
         with client.start_session() as s:
             s.with_transaction(create_and_insert)
 
-        # Outside a transaction we raise CollectionInvalid on existing colls.
-        with self.assertRaises(CollectionInvalid):
-            db.create_collection(coll.name)
+        # Outside a transaction we raise OperationFailure on existing colls.
+        if not client_context.is_topology_type(["sharded"]):
```
Comment (ShaneHarvey) [inline test/test_transactions.py:236]: `client_context.is_topology_type(["sharded"])` -> `client_context.is_mongos`

File: test/test_database.py [hunk 5]
```diff
@@ -132,7 +131,8 @@ def test_create_collection(self):
         db = Database(self.client, "pymongo_test")
 
         db.test.insert_one({"hello": "world"})
-        self.assertRaises(CollectionInvalid, db.create_collection, "test")
+        if not client_context.is_topology_type(["sharded"]):
+            self.assertRaises(OperationFailure, db.create_collection, "test")
```
Comment (ShaneHarvey) [inline test/test_database.py:135]: We should also test the behavior of `db.create_collection` on an existing collection on a sharded cluster. For example, there should be an `else:` here or we should have a test called `test_create_collection_existing`

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: doc/migrate-to-pymongo4.rst (hunk 1 above)
Comment (juliusgeo) [reply doc/migrate-to-pymongo4.rst:340]: I updated the docs to be more clear.

**Comment by juliusgeo** [comment]
Closing pull request because we do not know if we want to make this change yet. There is a variety of behavior on different server versions rendering this pull request non-functional/has a lot of hacking to ensure tests pass.

**Pull request closed without merging**